cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -105,4 +148,204 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+              return;
+            }
+          } else {
+            SupportBundleFiles supportBundleFiles = bundleGenerator.getFilesName(pipelineFile);
+            if (supportBundleFiles != null) {
+              supportBundleFilesList.add(supportBundleFiles);
+            }
+          }
+        }
+      }
+      responder.sendString(HttpResponseStatus.OK, GSON.toJson(supportBundleFilesList));
+      return;
+    }
+    File folderDirectory = new File(uuidFile, folderName);
+    File[] dataFiles =
+      folderDirectory.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+    if (dataFiles != null && dataFiles.length > 0) {
+      for (File dataFile : dataFiles) {
+        if (dataFile.getName().startsWith(dataFileName)) {
+          BufferedReader br = new BufferedReader(new FileReader(dataFile));
+          StringBuilder sb = new StringBuilder();
+          String line = br.readLine();
+          while (line != null) {
+            sb.append(line).append("\n");
+            line = br.readLine();
+          }
+          responder.sendString(HttpResponseStatus.OK, sb.toString());
+          return;
+        }
+      }
+    }
+    responder.sendString(HttpResponseStatus.NOT_FOUND,
+                         String.format("uuid: %s with pipeline file name: %s and data file name: %s not found", uuid,
+                                       folderName, dataFileName));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
+  public void appsExport(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId supportBundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(supportBundleEntityId, StandardPermission.GET);
+    String requestContent = request.content().toString(StandardCharsets.UTF_8);
+    if (requestContent == null || requestContent.length() == 0) {
+      throw new BadRequestException("Request body is empty.");
+    }
+    SupportBundleExportRequest supportBundleExportRequest =
+      GSON.fromJson(requestContent, SupportBundleExportRequest.class);
+    File tempDir = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR),
+                            cConf.get(Constants.SupportBundle.SUPPORT_BUNDLE_TEMP_DIR)).getAbsoluteFile();
+    DirUtils.mkdirs(tempDir);
+    java.nio.file.Path tmpPath = Files.createTempFile(tempDir.toPath(), "collect_support_bundle", ".zip");
+    List<String> listOfRequestFilePath = new ArrayList<>();
+    if (supportBundleExportRequest.getSupportBundleRequestFileList().getFiles().size() > 0) {
+      List<String> fileList = supportBundleExportRequest.getSupportBundleRequestFileList().getFiles();
+      fileList.iterator().forEachRemaining(files -> listOfRequestFilePath.add(files));
```

Comment:
Simplify to 

`listOfRequestFilePath.addAll(supportBundleExportRequest.getSupportBundleRequestFileList().getFiles());`